### PR TITLE
[SR-4917] Insert data into olap external table failed because range partition info missed  for partition table

### PR DESCRIPTION
**Pull request by wuleistarrocks**

### CODE REVIEW

File: gensrc/thrift/FrontendService.thrift [hunk 1]
```diff
@@ -752,18 +753,33 @@ struct TSchemaMeta {
...
+}
+
+struct TRangePartitionDesc {
+    1: optional i64 partition_id
+    2: optional binary start_key
+    3: optional binary end_key
 }
 
 struct TPartitionInfo {
     1: optional string type
```
Comment (imay) [inline gensrc/thrift/FrontendService.thrift:777]: Better to use enum as type

File: gensrc/thrift/FrontendService.thrift [hunk 2]
```diff
@@ -752,18 +753,33 @@ struct TSchemaMeta {
...
     2: optional map<i64, i16> replica_num_map
     3: optional map<i64, bool> in_memory_map
+    4: optional list<TColumnMeta> columns
+    5: optional map<i64, TRangePartitionDesc> partition_desc
+    6: optional map<i64, TDataProperty> data_property
```
Comment (imay) [inline gensrc/thrift/FrontendService.thrift:782]: StarRocks supprot multiple types of partition.
I think it is a good way to create a struct for each partition type. such as a TRangePartitionDesc, TSinglePartitionDesc. When `list partition` is supported, TListPartitionDesc can be supported.

File: gensrc/thrift/FrontendService.thrift [hunk 3]
```diff
@@ -752,18 +753,33 @@ struct TSchemaMeta {
...
+    4: optional i64 row_count
+    5: optional i64 rollup_index_id
+    6: optional i64 rollup_finished_version
+    7: optional TSchemaMeta schema_meta
+    8: optional list<TTabletMeta> tablets
+}
+
+struct TDataProperty {
+    1: Types.TStorageMedium storage_medium
+    2: i64 cold_time
```
Comment (imay) [inline gensrc/thrift/FrontendService.thrift:767]: optional

File: gensrc/thrift/FrontendService.thrift [hunk 4]
```diff
@@ -752,18 +753,33 @@ struct TSchemaMeta {
...
 struct TIndexMeta {
     1: optional i64 index_id
-    2: optional string index_state
-    3: optional i64 row_count
-    4: optional i64 rollup_index_id
-    5: optional i64 rollup_finished_version
-    6: optional TSchemaMeta schema_meta
-    7: optional list<TTabletMeta> tablets
+    2: optional i64 partition_id
+    3: optional string index_state
```
Comment (imay) [inline gensrc/thrift/FrontendService.thrift:757]: use enum other than string

File: fe/fe-core/src/main/java/com/starrocks/planner/OlapTableSink.java [hunk 5]
```diff
@@ -121,6 +121,7 @@ public void updateLoadId(TUniqueId newLoadId) {
     public void complete() throws UserException {
         TOlapTableSink tSink = tDataSink.getOlap_table_sink();
 
+        LOG.info("dst table name: {}, type: {}", dstTable.getName(), dstTable.getClass().getName());
```
[the remark this answers is not in the document]
Comment (wuleistarrocks) [reply fe/fe-core/src/main/java/com/starrocks/planner/OlapTableSink.java]: ok

**Review by imay — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Pull request merged**